File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/connectors/PyCheckiOOAuthConnector.java
package com.jetbrains.edu.python.learning.checkio.connectors;

import com.jetbrains.edu.learning.checkio.account.CheckiOAccount;
import com.jetbrains.edu.learning.checkio.connectors.CheckiOOAuthConnector;
import com.jetbrains.edu.python.learning.checkio.PyCheckiOSettings;
import com.jetbrains.edu.python.learning.checkio.utils.PyCheckiONames;
import com.jetbrains.edu.python.learning.checkio.utils.PyCheckiOOAuthBundle;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class PyCheckiOOAuthConnector extends CheckiOOAuthConnector {
  private final static String CLIENT_ID = PyCheckiOOAuthBundle.messageOrDefault("pyCheckioClientId", "");
  private final static String CLIENT_SECRET = PyCheckiOOAuthBundle.messageOrDefault("pyCheckioClientSecret", "");

  private PyCheckiOOAuthConnector() {
    super(CLIENT_ID, CLIENT_SECRET);
  }

  @Nullable
  @Override
  public CheckiOAccount getAccount() {
    return PyCheckiOSettings.INSTANCE.getAccount();
  }

  @Override
  public void setAccount(@Nullable CheckiOAccount account) {
    PyCheckiOSettings.INSTANCE.setAccount(account);
  }

  @NotNull
  @Override
  public String getOAuthServicePath() {
    return PyCheckiONames.PY_CHECKIO_OAUTH_SERVICE_PATH;
  }

  @NotNull
  @Override
  protected String getPlatformName() {
    return PyCheckiONames.PY_CHECKIO;
  }

  private static class Holder {
    private static final PyCheckiOOAuthConnector INSTANCE = new PyCheckiOOAuthConnector();
  }

  @NotNull
  public static PyCheckiOOAuthConnector getInstance() {
    return Holder.INSTANCE;
  }
}

File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/connectors/PyCheckiOOAuthConnector.kt
package com.jetbrains.edu.python.learning.checkio.connectors

import com.jetbrains.edu.learning.checkio.account.CheckiOAccount
import com.jetbrains.edu.learning.checkio.connectors.CheckiOOAuthConnector
import com.jetbrains.edu.learning.checkio.utils.CheckiONames
import com.jetbrains.edu.learning.checkio.utils.CheckiONames.PY_CHECKIO_PREFIX
import com.jetbrains.edu.python.learning.checkio.PyCheckiOSettings
import com.jetbrains.edu.python.learning.checkio.utils.PyCheckiOOAuthBundle

object PyCheckiOOAuthConnector : CheckiOOAuthConnector() {
  @get:Synchronized
  @set:Synchronized
  override var account: CheckiOAccount?
    get() = PyCheckiOSettings.getInstance().account
    set(account) {
      PyCheckiOSettings.getInstance().account = account
    }

  override val clientId: String = PyCheckiOOAuthBundle.value("pyCheckioClientId")

  override val clientSecret: String = PyCheckiOOAuthBundle.value("pyCheckioClientSecret")

  override val serviceName: String = getCheckiOServiceName(PY_CHECKIO_PREFIX.toLowerCase())

  override val platformName: String = CheckiONames.PY_CHECKIO
}